### PR TITLE
Update autoconf to fail if missing numa or m4

### LINKED ISSUE #6: configure should detect missing packages
configure detects some missing packages such as zip, but not others such as libnuma-dev.
This causes the build to fail during compilation.

### PR DESCRIPTION
**Pull request by jdekonin**
Fixes #6: configure should detect missing packages

Signed-off-by: Joe Dekoning <joe_dekoning@ca.ibm.com>

### CODE REVIEW

**Comment by mstoodle** [comment]
Thanks for this, @jdekonin . I like the idea to deal with these dependencies better.

I notice the changes are all in the common configuration. I am no expert in these files, so please forgive a naive question: do these changes need to be made on top of the files in the `common` directory, or could they be implemented in or "beside" the files we created in the `closed` directory?

**Comment by jdekonin** [comment]
@mstoodle, this last update I did most of the changes in the closed directory which does keep the configuration separate.

I left the help.m4 change as it was as I was a little hesitant in writing a separate and mostly identical solution to what openjdk has already implemented.

File: closed/autoconf/custom-hook.m4 [hunk 1]
```diff
@@ -37,12 +39,35 @@ AC_DEFUN_ONCE([CUSTOM_EARLY_HOOK],
...
   OPENJ9_PLATFORM_SETUP
   OPENJDK_VERSION_DETAILS
   OPENJ9_CONFIGURE_CUDA
   OPENJ9_THIRD_PARTY_REQUIREMENTS
 ])
 
+AC_DEFUN([OPENJ9_LIB_DETERMINE_DEPENDENCIES],
+[
+  # Check if numa is needed
+  if test "x$OPENJDK_TARGET_OS" = xlinux && test "x$OPENJDK_BUILD_CPU_ARCH" != xs390; then
```
Comment (mstoodle) [inline closed/autoconf/custom-hook.m4]: I know we don't have ARM builds active yet, but this condition seems a bit too open ended to me. Perhaps better to explicitly list the platforms where it should be done?

File: common/autoconf/generated-configure.sh [hunk 2]
```diff
@@ -1100,6 +1100,7 @@ infodir
 docdir
 oldincludedir
 includedir
+runstatedir
```
Comment (mstoodle) [inline common/autoconf/generated-configure.sh]: what is this runstatedir thing? Maybe I missed it, but it doesn't seem to be used by the other changes in this PR?

File: closed/autoconf/custom-hook.m4 (hunk 1 above)
Comment (jdekonin) [reply closed/autoconf/custom-hook.m4]: Openj9 supports numa on x86 and ppc architectures, so I updated the condition accordingly.

File: common/autoconf/generated-configure.sh (hunk 2 above)
Comment (jdekonin) [reply common/autoconf/generated-configure.sh]: I'm not sure why anything in this file changed other than the timestamp which is the only change in the new commit pushed up.

**Comment by jdekonin** [comment]
There is going to be a *generated-configure.sh conflict as the change is.  Once #25 is merged I can update this one.

**Comment by jdekonin** [comment]
Rebased and tested on the latest and everything works as expected.

**Comment by jdekonin** [comment]
Rebased and tested on the latest and everything works as expected.  Can I please get a review or merge please?

**Comment by jdekonin** [comment]
@keithc-ca Can you please review?

File: closed/autoconf/custom-hook.m4 [hunk 3]
```diff
@@ -37,12 +37,33 @@ AC_DEFUN_ONCE([CUSTOM_EARLY_HOOK],
...
 ])
 
+AC_DEFUN([OPENJ9_BASIC_SETUP_FUNDAMENTAL_TOOLS],
+[
+  BASIC_REQUIRE_PROGS(M4, m4)
+])
+
+AC_DEFUN_ONCE([OPENJ9_CONFIGURE_NUMA],
+[
+  if test "x$OPENJDK_TARGET_CPU_ARCH" = xx86 || "x$OPENJDK_TARGET_CPU_ARCH" = xppc; then
```
Comment (keithc-ca) [inline closed/autoconf/custom-hook.m4]: I don't think this is what you meant: you probably want `-o` instead of `||`.

Comment (jdekonin) [reply closed/autoconf/custom-hook.m4]: Change made.

**Comment by jdekonin** [comment]
I believe the contents of that file was not up-to-date.  Same comment was made on #25 when I made that change which was reverted until doc could be updated.

**Comment by jdekonin** [comment]
closed/autoconf/generated-configure.sh is large because I am incorporating an autoconf macro that checks the existence of the header files as well as the fact that the lines have all shifted with the addition of the m4 cmd.

Most recent is a rebase on the current, plus an addition that limits the numa requirement to linux based x86 and ppc operating system.

**Comment by keithc-ca** [comment]
I believe @jdekonin has addressed comments from @mstoodle and this looks good to me.

**Pull request merged**